src/components/NostoProduct/store.ts

```diff
@@ -50,11 +61,39 @@ export function createStore(element: NostoProduct) {
     if (totalSelection === state.optionGroupCount) {
       const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
       if (selectedSkuIds.length === 1) {
-        selectSkuId(selectedSkuIds[0])
+        const skuId = selectedSkuIds[0]
+
+        const elementWithImage = Object.values(state.selectedElements).find(
+          el => el?.getAttribute("n-skus")?.split(",").includes(skuId) && el?.hasAttribute("ns-img")
+        )
+
+        selectSkuId(skuId, elementWithImage)
       }
     }
   }
 
+  function setSkuImages(image: string, altImage?: string) {
+    if (!state.selectedSkuId) return
+    state.skuImage = image
+    notify("skuImage", image)
+
+    if (altImage) {
+      state.skuAltImage = altImage
+      notify("skuAltImage", altImage)
+
+      const preload = new Image()
```

Comment:
should we leave this to be handled by template level logic?
or maybe via NostoSkuOptions?
at least the store shouldn't need to deal with this